### PR TITLE
Use CLI for cleaning environments

### PR DESCRIPTION
**Pull request by karola312**
Use CLI in order to clean environments (release environments and master weekly, automation dashboard every night)
### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets file (instructions in [contribution guide](https://github.com/saleor/saleor-dashboard/blob/main/.github/CONTRIBUTING.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [ ] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=2

### CODE REVIEW

File: .github/workflows/clean-envs.yml [hunk 1]
```diff
@@ -24,18 +24,22 @@ jobs:
           cd .github/workflows
           npm ci
 
+      - name: Write config file
+        id: write-config-file
+        env:
+          ACCESS_TOKEN: ${{ secrets.STAGING_TOKEN }}
+        run: echo '{"token":"Token ${{ env.ACCESS_TOKEN }}","telemetry":"false","organization_slug":"saleor","organization_name":"Saleor"}' > ~/.config/saleor.json
```
Comment (NyanKiyoshi) [inline .github/workflows/clean-envs.yml:31]: Use `jq` to build JSON from arbitrary values:

```suggestion
        run: jq --null-input --arg token "Token $ACCESS_TOKEN" '{"token": $token }", "telemetry": "false", "organization_slug": "saleor", "organization_name": "Saleor"}' > ~/.config/saleor.json
```

File: .github/workflows/cleanEnvironments.js [hunk 2]
```diff
@@ -47,73 +46,61 @@ program
...
-        "Content-Type": "application/json;charset=UTF-8",
-      },
-    },
-  );
-  return await response.json();
+async function getEnvironments() {
+  const result = await exec("npx saleor env list --json", {
+    serialization: "json",
+  });
+  const envList = result.stdout.replace("\x1B[?25l\x1B[?25h", "");
```
Comment (NyanKiyoshi) [inline .github/workflows/cleanEnvironments.js:63]: Can you add comment on what you are doing here? (the ascii escape part: `\x1B[?25l\x1B[?25h` - which is probably hiding and showing the cursor)

Does saleor-cli have a pipe mode or CI mode so we don't need to remove the ASCII escapes?

File: .github/workflows/cleanEnvironments.js [hunk 3]
```diff
@@ -47,73 +46,61 @@ program
...
+async function getSnapshotsForRestore(allEnvironments) {
+  let allBackups = [];
+  await Promise.all(
+    allEnvironments.map(async environment => {
+      const result = await exec(
+        `npx saleor backup list --json --organization="saleor" --environment="${environment.key}"`,
+        {
+          serialization: "json",
+        },
+      );
```
Comment (NyanKiyoshi) [inline .github/workflows/cleanEnvironments.js:94]: Use `child_process.execFile(file[, args][, options][, callback])` instead, `child_process.exec(command[, options][, callback])` is not designed for handling unsanitized arguments

File: .github/workflows/tests-nightly.yml [hunk 4]
```diff
@@ -64,11 +64,16 @@ jobs:
           cd .github/workflows
           npm ci
 
+      - name: Write config file
+        id: write-config-file
+        env:
+          ACCESS_TOKEN: ${{ secrets.STAGING_TOKEN }}
+        run: echo '{"token":"Token ${{ env.ACCESS_TOKEN }}","telemetry":"false","organization_slug":"saleor","organization_name":"Saleor"}' > ~/.config/saleor.json
```
Comment (NyanKiyoshi) [inline .github/workflows/tests-nightly.yml:71]: Use `jq` instead

**Review by NyanKiyoshi — CHANGES_REQUESTED** [review]

File: .github/workflows/cleanEnvironments.js [hunk 5]
```diff
@@ -14,29 +13,29 @@ let warningMessage = "";
...
     );
+    const environments = await getEnvironments();
     const environmentsToClean = await getEnvironmentsToClean(
-      token,
+      environments,
       environmentsToCleanRegex,
     );
-    const snapshotsForRestore = await getSnapshotsForRestore(token);
+    const snapshotsForRestore = await getSnapshotsForRestore(environments);
+    console.log(snapshotsForRestore);
```
Comment (andrzejewsky) [inline .github/workflows/cleanEnvironments.js:30]: mistake?

**Pull request closed without merging**